app/code/Magento/SalesRule/Model/Validator.php

```diff
@@ -409,6 +403,33 @@ public function initTotals($items, Address $address)
         return $this;
     }
 
+    /**
+     * Determine if quote item is valid for a given sales rule
+     *
+     * @param AbstractItem $item
+     * @param Rule $rule
+     * @return bool
+     */
+    protected function isValidItemForRule(
+        AbstractItem $item,
+        Rule $rule
+    ) {
+        /** @var AbstractItem $item */
```

Comment:
We don’t need this phpdoc block as we have this type defined in method arguments

Reply:
Ah - thanks - forgot to remove that one.